tests/system/src/test/java/test/robot/javafx/web/PointerEventTest.java

```diff
@@ -0,0 +1,229 @@
+import java.util.concurrent.TimeUnit;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+
+import static org.junit.Assert.fail;
+
+/*
+ * Tests for validating the buttons property received in "pointermove" event,
+ * when various mouse buttons are pressed and dragged
+ * testLeftButtonDrag           - Press the Mouse Left Button and drag
+ * testRightButtonDrag          - Press the Mouse Right Button and drag
+ * testMiddleButtonDrag         - Press the Mouse Middle Button and drag
+ * testLeftMiddleButtonDrag     - Press Left and Middle Buttons and drag
+ * testMiddleRightButtonDrag    - Press Middle and Right Buttons and drag
+ * testLeftRightButtonDrag      - Press Left and Right Buttons and drag
+ * testLeftMiddleRightButtonDrag- Press Left, Middle and Right Buttons and drag
+ */
+
+public class PointerEventTest {
+
+    private static final int LEFT_BUTTON_DRAG = 1;
+    private static final int MIDDLE_BUTTON_DRAG = 4;
+    private static final int RIGHT_BUTTON_DRAG = 2;
+    private static final int SCENE_WIDTH = 250;
+    private static final int SCENE_HEIGHT = SCENE_WIDTH;
+    // Sleep time between mouseMove and element access
+    private static final int SLEEP_TIME = 500;
+
+    private final int DRAG_DISTANCE = 15;
+    private final int DX = 125;
+    private final int DY = 125;
+
+    private static CountDownLatch loadLatch;
+    private static CountDownLatch startupLatch;
+
+    static Document document;
+    static Element element;
+    static Robot robot;
+    static WebView webView;
+    static WebEngine webEngine;
+
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    private String buttonMask;
+    private int result = 0;
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            robot = new Robot();
+            stage = primaryStage;
+            stage.setTitle("Mouse Drag Test");
+            webView = new WebView();
+            webEngine = webView.getEngine();
+            String URL =  this.getClass().getResource("pointerEvent.html").toString();
+            webEngine.load( URL);
+            webView.getEngine().getLoadWorker().stateProperty().addListener((ov, o, n) -> {
+                if (n == Worker.State.SUCCEEDED) {
+                    document = webEngine.getDocument();
+                    element = document.getElementById("buttonPressed");
+                    loadLatch.countDown();
+                }
+            });
+            scene = new Scene(new StackPane(webView), SCENE_WIDTH, SCENE_HEIGHT);
+            stage.setScene(scene);
+            stage.setAlwaysOnTop(true);
+            Platform.runLater(startupLatch::countDown);
+            stage.show();
+        }
+    }
+
+    public String mouseButtonDrag(MouseButton... button) {
```

Comment:
I recommend changing the name of the parameter to `buttonArray` (or `buttonArr` or just `buttons` if you want it shorter), since this varargs parameter is an array of potentially more than one button.

Reply:
changed to buttons